File: Backend/backend/src/main/java/com/example/backend/dto/event/EventDto.java
package com.example.backend.dto.event;

import java.time.Instant;

public class EventDto {
    private Long id;
    private Long userId;
    private Long templateId;
    private String name;
    private String description;
    private String imageUrl;
    private Instant startTime;
    private Instant endTime;

    public EventDto() {}

    public Long getId()                     { return id; }
    public void setId(Long id)              { this.id = id; }

    public Long getUserId()                     { return userId; }
    public void setUserId(Long userId)          { this.userId = userId; }

    public Long getTemplateId()                 { return templateId; }
    public void setTemplateId(Long templateId)  { this.templateId = templateId; }

    public String getName()                     { return name; }
    public void setName(String name)            { this.name = name; }

    public String getDescription()              { return description; }
    public void setDescription(String desc)     { this.description = desc; }

    public String getImageUrl()                 { return imageUrl; }
    public void setImageUrl(String imageUrl)    { this.imageUrl = imageUrl; }

    public Instant getStartTime()               { return startTime; }
    public void setStartTime(Instant startTime) { this.startTime = startTime; }

    public Instant getEndTime()                 { return endTime; }
    public void setEndTime(Instant endTime)     { this.endTime = endTime; }
}

File: KotlinClient/app/src/main/java/com/example/kotlinclient/api_client/dto/EventDto.kt
package com.example.kotlinclient.api_client.dto

data class EventDto(
    val id: Long,
    val userId: Long,
    val templateId: Long?,
    val name: String?,
    val description: String?,
    val imageUrl: String?,
    val startTime: String,   // ISO-8601 Instant → "2026-06-01T10:00:00Z"
    val endTime: String
)

data class EventCreateRequest(
    val templateId: Long?,
    val name: String?,
    val description: String?,
    val imageUrl: String?,
    val startTime: String,
    val endTime: String
)

data class EventUpdateRequest(
    val name: String?,
    val description: String?,
    val imageUrl: String?,
    val startTime: String?,
    val endTime: String?
)
